ocaml/typing/env.ml

```diff
@@ -1496,12 +1498,16 @@ let find_modtype_expansion path env =
 let rec is_functor_arg path env =
   match path with
     Pident id ->
-      begin try Ident.find_same id env.functor_args; true
-      with Not_found -> false
-      end
+      Option.is_some (Ident.find_same_opt id env.functor_args)
```

Comment:
I agree the `_opt` version of the function is an improvement over the raising version, but I would discourage making "cleanup" changes to unrelated parts of the compiler as part of this branch; in general it can make the rebase harder.